lib/elixir/test/elixir/kernel/has_unquote_test.exs

```diff
@@ -0,0 +1,122 @@
+Code.require_file("../test_helper.exs", __DIR__)
+
+defmodule Kernel.HasUnquoteTest do
```

Comment:
Feel free to suggest a better place for these tests.

I know `has_unquotes/1` is purely internal, but given its complexity and edge cases, unit-testing feels highly valuable to validate and reason about its expected behavior.

Reply:
I would move to kernel/quote_test.exs :)

Reply:
Perfect! Done.